### PR TITLE
[PAY-1456] Fix followee-gated create chat crash

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
- Moved the buttons into the `switch` statement, which prevents crashes when we encounter a case that wasn't in the `actionToButtonsMap`.
- Fix the followee-gated case in `getCanCreateChat` selector lol.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Local ios stage
Simulator Screen Shot - iPhone 14 Pro - 2023-06-14 at 17 55 47

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/common/src/store/pages/chat/selectors.ts [hunk 1]
```diff
@@ -269,7 +269,7 @@ export const getCanCreateChat = createSelector(
         action = ChatPermissionAction.NONE
       } else if (
         userPermissions.permits === ChatPermission.FOLLOWEES &&
-        !user?.does_current_user_follow
+        !user?.does_follow_current_user
```
Comment (rickyrombo) [inline packages/common/src/store/pages/chat/selectors.ts:272]: 🤔 wow how did this go unseen for so long

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-perms-crash

File: packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx [hunk 2]
```diff
@@ -105,129 +105,149 @@ const useStyles = makeStyles(({ spacing, typography, palette }) => ({
   }
 }))
 
-const mapActionToContent = (
+const useGetDrawerContent = (
```
Comment (rickyrombo) [inline packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx:107]: is this not just a component with props? 👀

**Review by rickyrombo — APPROVED** [review]

File: packages/common/src/store/pages/chat/selectors.ts (hunk 1 above)
Comment (dharit-tan) [reply packages/common/src/store/pages/chat/selectors.ts:272]: i lol'd when i found it

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/rt-perms-crash

File: packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx (hunk 2 above)
Comment (dharit-tan) [reply packages/mobile/src/components/inbox-unavailable-drawer/InboxUnavailableDrawer.tsx:107]: wow you're so right

**Review by piazzatron — APPROVED** [review]
Oh man nice find 😂

**Pull request merged**